relay-metrics/src/aggregator.rs

```diff
@@ -748,52 +721,107 @@ impl FromMessage<BucketCountInquiry> for Aggregator {
 /// Metrics are uniquely identified by the combination of their name, type and unit. It is allowed
 /// to send metrics of different types and units under the same name. For example, sending a metric
 /// once as set and once as distribution will result in two actual metrics being recorded.
-///
-/// # Flushing
-///
-/// Buckets are flushed to a receiver after their time window and a grace period have passed.
-/// Metrics with a recent timestamp are given a longer grace period than backdated metrics, which
-/// are flushed after a shorter debounce delay. See [`AggregatorConfig`] for configuration options.
-///
-/// Internally, the aggregator maintains a continuous flush cycle every 100ms. It guarantees that
-/// all elapsed buckets belonging to the same [`ProjectKey`] are flushed together.
-///
-/// Receivers must implement a handler for the [`FlushBuckets`] message.
-pub struct AggregatorService {
+pub struct Aggregator {
     name: String,
     config: AggregatorConfig,
     buckets: HashMap<BucketKey, QueuedBucket>,
-    receiver: Option<Recipient<FlushBuckets, NoResponse>>,
-    state: AggregatorState,
     cost_tracker: CostTracker,
 }
 
-impl AggregatorService {
-    /// Create a new aggregator and connect it to `receiver`.
-    ///
-    /// The aggregator will flush a list of buckets to the receiver in regular intervals based on
-    /// the given `config`.
-    pub fn new(
-        config: AggregatorConfig,
-        receiver: Option<Recipient<FlushBuckets, NoResponse>>,
-    ) -> Self {
-        Self::named("default".to_owned(), config, receiver)
+impl Aggregator {
+    /// Returns the name of the aggregator.
+    pub fn name(&self) -> &str {
+        self.name.as_str()
     }
 
-    /// Like [`Self::new`], but with a provided name.
-    pub(crate) fn named(
-        name: String,
-        config: AggregatorConfig,
-        receiver: Option<Recipient<FlushBuckets, NoResponse>>,
-    ) -> Self {
-        Self {
-            name,
-            config,
-            buckets: HashMap::new(),
-            receiver,
-            state: AggregatorState::Running,
-            cost_tracker: CostTracker::default(),
+    /// Returns the config of the aggregator.
+    pub fn config(&self) -> &AggregatorConfig {
+        &self.config
+    }
+
+    /// Returns the number of buckets in the aggregator.
+    pub fn bucket_qty(&self) -> usize {
+        self.buckets.len()
+    }
+
+    /// Returns `true` if the cost trackers value is larger than the given max cost.
+    pub fn totals_cost_exceeded(&self, max_total_cost: Option<usize>) -> bool {
+        self.cost_tracker.totals_cost_exceeded(max_total_cost)
+    }
+
+    /// Pop and return the buckets that are eligible for flushing out according to bucket interval.
+    ///
+    /// Note that this function is primarily intended for tests.
+    pub fn pop_flush_buckets(&mut self, force: bool) -> HashMap<ProjectKey, Vec<HashedBucket>> {
```

Comment:
alternatively, could have a `force_pop_flush_bucket` and `pop_flush_bucket` that each call an inner function with/without the force flag